query-connector/src/app/fhir-servers/page.tsx

```diff
@@ -0,0 +1,392 @@
+      };
+    }
+  };
+
+  const handleTestConnection = async () => {
+    const result = await testFhirConnection(serverUrl);
+    setConnectionStatus(result.success ? "success" : "error");
+    setErrorMessage(result.error);
+  };
+
+  const handleSave = async () => {
+    // First test the connection
+    const connectionResult = await testFhirConnection(serverUrl);
+
+    if (modalMode === "create") {
+      const result = await insertFhirServer(
+        serverName,
+        serverUrl,
+        connectionResult.success,
+      );
+
+      if (result.success) {
+        getFhirServerConfigs(true).then((servers) => {
+          setFhirServers(servers);
+        });
+        handleCloseModal();
+      } else {
+        setConnectionStatus("error");
+        setErrorMessage(result.error);
+      }
+    } else if (selectedServer) {
+      const result = await updateFhirServer(
+        selectedServer.id,
+        serverName,
+        serverUrl,
+        connectionResult.success,
+      );
+
+      if (result.success) {
+        getFhirServerConfigs(true).then((servers) => {
+          setFhirServers(servers);
+        });
+        handleCloseModal();
+      } else {
+        setConnectionStatus("error");
+        setErrorMessage(result.error);
+      }
+    }
+  };
+
+  const handleDeleteServer = async () => {
+    if (!selectedServer) return;
+
+    const result = await deleteFhirServer(selectedServer.id);
+
+    if (result.success) {
+      // Refresh the server list after successful deletion
+      getFhirServerConfigs(true).then((servers) => {
+        setFhirServers(servers);
+      });
+      handleCloseModal();
+    } else {
+      setConnectionStatus("error");
+      setErrorMessage(result.error);
+    }
+  };
+
+  const getModalButtons = () => {
+    const buttons = [
+      {
+        text: modalMode === "create" ? "Add server" : "Save changes",
+        type: "submit" as const,
+        id: "modal-save-button",
+        className: "usa-button",
+        onClick: handleSave,
+      },
+      {
+        text: "Test connection" as string | JSX.Element,
+        type: "button" as const,
+        id: "modal-test-connection-button",
+        className: "usa-button usa-button--outline",
+        onClick: handleTestConnection,
+      },
+      {
+        text: "Cancel",
+        type: "button" as const,
+        id: "modal-cancel-button",
+        className: "usa-button usa-button--outline shadow-none",
+        onClick: handleCloseModal,
+      },
+    ];
+
+    // Add delete button only in edit mode
+    if (modalMode === "edit") {
+      buttons.push({
+        text: "Delete",
+        type: "button" as const,
+        id: "modal-delete-button",
+        className: "usa-button usa-button--secondary",
+        onClick: handleDeleteServer,
+      });
+    }
+
+    // Update test connection button if successful
+    if (connectionStatus === "success") {
+      buttons[1].text = (
+        <>
+          <Icon.Check
+            size={3}
```

Comment:
I'm still seeing the button size jump after the connection check, which I think is caused by the checkmark icon? Bumping this down to size 2 keeps it in line with the line height of the text

Reply:
(or decrease the top/bottom padding to accommodate, since I see the linter yelling about 2 not being an acceptable value here)